File: backend/src/main/java/com/slembers/alarmony/alarm/dto/AlarmDto.java
package com.slembers.alarmony.alarm.dto;

import com.slembers.alarmony.global.util.CommonMethods;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;

import java.util.List;

@Getter
@NoArgsConstructor
@AllArgsConstructor
public class AlarmDto {

    private Long alarmId;

    private String title;

    private String content;

    private int hour;

    private int minute;

    private List<Boolean> alarmDate;

    /**
     *
     * @param alarmId 알람 아이디
     * @param title 제목
     * @param content 내용
     * @param hour 시간
     * @param minute 분
     * @param alarmDate 알람 요일 정보
     */
    public AlarmDto(Long alarmId, String title, String content, int hour, int minute, String alarmDate) {
        this.alarmId = alarmId;
        this.title = title;
        this.content = content;
        this.hour = hour;
        this.minute = minute;
        this.alarmDate = CommonMethods.changeStringToBooleanList(alarmDate);
    }
}

File: android/app/src/main/java/com/slembers/alarmony/feature/alarm/AlarmDto.kt
package com.slembers.alarmony.feature.alarm

import android.os.Parcelable
import com.google.gson.annotations.SerializedName
import kotlinx.android.parcel.Parcelize

data class AlarmDto(
    @SerializedName("alarmId") val alarmId : Long,
    @SerializedName("title") val title : String,
    @SerializedName("hour") val hour : Int,
    @SerializedName("minute") val minute : Int,
    @SerializedName("alarmDate") val alarmDate : List<Boolean>,
    @SerializedName("soundName") val soundName : String,
    @SerializedName("soundVolume") val soundVolume : Int,
    @SerializedName("vibrate") val vibrate : Boolean,
    @SerializedName("host") val host : Boolean,
    @SerializedName("content") var content : String
) {
    companion object {
        fun toDto(alarm: Alarm): AlarmDto {
            val alarmDto = AlarmDto(
                alarm.alarmId,
                alarm.title,
                alarm.hour,
                alarm.minute,
                alarm.alarmDate,
                alarm.soundName,
                alarm.soundVolume,
                alarm.vibrate,
                alarm.host,
                alarm.content
            )
            return alarmDto
        }
    }
}